katcp/server.py

```diff
@@ -2237,13 +2250,25 @@ def request_sensor_sampling(self, req, msg):
             ?sensor-sampling cpu.power.on
             !sensor-sampling ok cpu.power.on none
 
-            ?sensor-sampling cpu.power.on period 500
-            !sensor-sampling ok cpu.power.on period 500
+            ?sensor-sampling cpu.power.on period 0.5
+            !sensor-sampling ok cpu.power.on period 0.5
+
+            if BULK_SET_SENSOR_SAMPLING is enabled then:
+
+            ?sensor-sampling cpu.power.on,fan.speed
+            !sensor-sampling fail Cannot\_query\_multiple\_sensors
+
+            ?sensor-sampling cpu.power.on,fan.speed period 0.5
+            !sensor-sampling ok cpu.power.on,fan.speed period 0.5
 
         """
+        if self.PROTOCOL_INFO.bulk_set_sensor_sampling and len(msg.arguments) > 1:
+            handler = self._handle_bulk_set_sensor_sampling
+        else:
+            handler = self._handle_sensor_sampling
```

Comment:
This implementation will fail when one requests multiple sensors without passing a strategy.
eg: 
```bash
> ?sensor-sampling bandwidth,n-chans
< !sensor-sampling fail Unknown sensor name: bandwidth,n-chans. 
```

This should reply with the last known value or strategy for the sensors passed.

Reply:
This request should fail, according to the guidelines you cannot query the strategies for multiple sensors.  But we could improve the error message, if bulk sensor sampling is enabled.